bakery/src/pipelines/cops.js

```diff
@@ -223,9 +233,15 @@ const pipeline = (env) => {
     on_success: reportToOutputProducerPdf(Status.SUCCEEDED, {
       pdf_url: 'artifacts/pdf_url'
     }),
-    on_failure: reportToOutputProducerPdf(Status.FAILED),
-    on_error: reportToOutputProducerPdf(Status.FAILED),
-    on_abort: reportToOutputProducerPdf(Status.FAILED)
+    on_failure: reportToOutputProducerPdf(Status.FAILED, {
```

Comment:
I'm not sure if this is possible to address if so, but if the put to S3 fails, will the error message contents end up coming from `taskBuildPdf`?

Reply:
Oh, good thinking. I will check on that.

Reply:
It sure does.

Reply:
I just pushed a change that adds a task that can be used to override the common log before non-task steps. It's a bit odd, and doesn't capture the specific failure, but it can print a generic message. Might be the best we can do, since I'm not really sure of any way to actually capture the output from a put step.

Reply:
Cool, good idea. Even if the task output details aren't conveyed at least the generic error message will give the user an accurate idea of why the job failed.